File: resultcapture-service/src/main/java/com/devotion/capture/service/QuestionAnswerRepository.java
package com.devotion.capture.service;

import com.devotion.capture.model.QuestionAnswer;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface QuestionAnswerRepository extends MongoRepository<QuestionAnswer, String> {

    List<QuestionAnswer> findBySurveyIdAndQuestionId(String surveyId, Integer questionId);

}

File: resultcapture-service/src/main/kotlin/com/devotion/capture/service/QuestionAnswerRepository.kt
package com.devotion.capture.service

import com.devotion.capture.model.QuestionAnswer
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface QuestionAnswerRepository : MongoRepository<QuestionAnswer, String> {

    fun findBySurveyIdAndQuestionId(surveyId: String, questionId: Int?): List<QuestionAnswer>

}


